src/compile/scale.ts

```diff
@@ -684,20 +677,85 @@ export function rangeMixins(scale: Scale, model: Model, channel: Channel):
   throw new Error(`Scale range undefined for channel ${channel}`);
 }
 
+function sizeRangeMin(mark: Mark, zero: boolean, markConfig: MarkConfig, scaleConfig: ScaleConfig) {
+  if (zero) {
+    return 0;
+  }
+  switch (mark) {
+    case 'bar':
+      return scaleConfig.minBarSize !== undefined ? scaleConfig.minBarSize : markConfig.barThinSize;
+    case 'tick':
+      return scaleConfig.minTickSize;
+    case 'rule':
+      return scaleConfig.minRuleSize;
+    case 'text':
+      return scaleConfig.minTextSize;
+    case 'point':
+    case 'square':
+    case 'circle':
+      return scaleConfig.minPointSize;
```

Comment:
This means that the min size is not 0 by default even if the smallest domain value is 0. I'm okay with that but just want to acknowledge it.